storage/redis_controller.go

```diff
@@ -41,8 +40,6 @@ func (rc *RedisController) DisableRedis(setRedisDown bool) {
 	rc.disableRedis.Store(false)
 	rc.redisUp.Store(false)
 
-	rc.reconnect <- struct{}{}
```

Comment:
So, the intent of this was to explicitly trigger reload when RedisController goes from Disabled to Enabled. I think the intent was to simulate redis going offline, reconnecting, and triggering a "reload" event when that happens. This is instant, otherwise we have to wait for ticker.C (time.Second).

Reply:
@titpetric Added it back but after the `rc.WaitConnect(ctx)`. If not, we can be triggering reconnection when Redis is down.

Reply:
That works! Good :)